### PR TITLE
add view department command

### LINKED ISSUE #103: Add View Department command
**As a user, I can view department of company, so that I can get summarised information on the department and its associated employees.**

The view department command will enable users to view all the persons under a department, as well as a summary of information on the department such as:

- All employee information
- Number of employees
- Department ratings

### PR DESCRIPTION
**Pull request by ngshijun**
Resolves #103

### CODE REVIEW

File: src/main/java/coydir/logic/commands/CommandResult.java [hunk 1]
```diff
@@ -23,6 +23,10 @@ public class CommandResult {
 
     private int viewIndex;
 
+    private String department;
+
+    private final boolean viewDepartment;
+
```
Comment (prit3010) [inline src/main/java/coydir/logic/commands/CommandResult.java:30]: Empty Line

File: src/main/java/coydir/logic/commands/ViewDepartmentCommand.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    private final String d;
+
+    public ViewDepartmentCommand(String d) {
+        this.d = d;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        if (Department.isValidDepartment(d)) {
```
Comment (prit3010) [inline src/main/java/coydir/logic/commands/ViewDepartmentCommand.java:32]: I think this check is redundant. What do you think?

File: src/main/java/coydir/logic/parser/DatabaseParser.java [hunk 3]
```diff
@@ -88,6 +92,7 @@ public Command parseCommand(String userInput) throws ParseException {
         case DeleteLeaveCommand.COMMAND_WORD:
             return new DeleteLeaveCommandParser().parse(arguments);
 
+
```
Comment (prit3010) [inline src/main/java/coydir/logic/parser/DatabaseParser.java:95]: Extra Line

File: src/main/java/coydir/logic/parser/ViewDepartmentCommandParser.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    public ViewDepartmentCommand parse(String args) throws ParseException {
+        try {
+            Department department = ParserUtil.parseDepartment(args);
+            return new ViewDepartmentCommand(department.toString());
+        } catch (ParseException pe) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, ViewDepartmentCommand.MESSAGE_USAGE), pe);
+        }
+    }
+
```
Comment (prit3010) [inline src/main/java/coydir/logic/parser/ViewDepartmentCommandParser.java:28]: Line 28 Extra Line

**Review by chao890 — APPROVED** [review]
LGTM

**Review by Kok-je — COMMENTED** [review]
Overall looks good, just some minor errors with documentation and styling.

**Review by prit3010 — COMMENTED** [review]
Looks good! Can remove the extra lines and can check if the isValidDepartment in ViewDepartmentCommand is necessary

**Pull request merged**